### ISSUE TITLE
Otto compile stuck

### ISSUE THREAD

**Issue by quim3ra** [hashicorp/otto#178]
I was following the getting starting guide.

I cloned the repository and execute otto compile and otto it's stuck.

```
C:\dev\otto-getting-started>otto compile
==> Loading Appfile...
==> No Appfile found! Detecting project information...
    No Appfile was found. If there is no Appfile, Otto will do its best
    to detect the type of application this is and set reasonable defaults.
    This is a good way to get started with Otto, but over time we recommend
    writing a real Appfile since this will allow more complex customizations,
    the ability to reference dependencies, versioning, and more.
==> Fetching all Appfile dependencies...
==> Compiling...
    Application:    otto-getting-started (ruby)
    Project:        otto-getting-started
    Infrastructure: aws (simple)

    Compiling infra...
    Compiling foundation: consul

```

I'm using Windows 7 64 bits.

**Comment by songinyourmind** [comment]
Same here for me. I'm on a mac though (Yosemite).

**Comment by justincampbell** [comment]
@quim3ra @songinyourmind can you Gist output of `OTTO_LOG=1 otto dev`?

**Comment by songinyourmind** [comment]
<script src="https://gist.github.com/songinyourmind/4cdfcabd93de03b90598.js"></script>

**Comment by songinyourmind** [comment]
https://gist.github.com/4cdfcabd93de03b90598.git

**Comment by songinyourmind** [comment]
https://gist.github.com/songinyourmind/4cdfcabd93de03b90598

**Comment by quim3ra** [comment]
Here you have it:

https://gist.github.com/quim3ra/6aacedd7b097d781f4cd

**Comment by johnyesberg** [comment]
I have a similar problem. It was working a couple of days ago - at least up to running the demo on a local VM. I tried to do a clean start by deleting the otto-getting-started and the .otto.d. But now I can't get past ```otto compile```. The end of my log is similar:

```
2015/10/05 20:11:42 [DEBUG] Copying all assets: data/common/app-build => C:\Users\John\projects\otest\otto-getting-started\.otto\compiled\foundation-consul\app-build
2015/10/05 20:11:42 [DEBUG] Copying asset: main.sh.tpl
2015/10/05 20:11:42 [DEBUG] Copying asset: upstart.conf.tpl
2015/10/05 20:11:42 [DEBUG] Copying asset: app-deploy
2015/10/05 20:11:42 [DEBUG] Copying all assets: data/common/app-deploy => C:\Users\John\projects\otest\otto-getting-started\.otto\compiled\foundation-consul\app-deploy
2015/10/05 20:11:42 [DEBUG] Copying asset: main.sh.tpl
2015/10/05 20:11:42 [DEBUG] Copying all assets: data/aws-simple => C:\Users\John\projects\otest\otto-getting-started\.otto\compiled\foundation-consul
2015/10/05 20:11:42 [DEBUG] Copying asset: deploy
2015/10/05 20:11:42 [DEBUG] Copying all assets: data/aws-simple/deploy => C:\Users\John\projects\otest\otto-getting-started\.otto\compiled\foundation-consul\deploy
2015/10/05 20:11:42 [DEBUG] Copying asset: main.tf
2015/10/05 20:11:42 [DEBUG] Copying asset: variables.tf
2015/10/05 20:11:42 [DEBUG] reading IP, cache path: C:\Users\John\projects\otest\otto-getting-started\.otto\data\dev_ip
2015/10/05 20:11:42 [DEBUG] no ip cache found, getting new IP
2015/10/05 20:11:42 [DEBUG] ignoring non IPv4 addr: fe80::ed2e:cddf:fef1:9be8/28
2015/10/05 20:11:42 [DEBUG] occupied private IP space: 192.168.1.5/16
2015/10/05 20:11:42 [DEBUG] ignoring non IPv4 addr: fe80::f117:3b71:3bf2:3398/28
2015/10/05 20:11:42 [DEBUG] occupied private IP space: 192.168.1.6/16
2015/10/05 20:11:42 [DEBUG] ignoring non IPv4 addr: fe80::90dd:a2c1:652f:9f4f/28
2015/10/05 20:11:42 [DEBUG] occupied private IP space: 192.168.56.1/16
2015/10/05 20:11:42 [DEBUG] ignoring non IPv4 addr: fe80::240f:769d:6d0a:2f56/28
2015/10/05 20:11:42 [DEBUG] occupied private IP space: 172.16.1.1/16
```

**Comment by johnyesberg** [comment]
It seems that this problem may be related to the availability of appropriate IP addresses. I was able to fix this problem by looking at my network adaptor settings, in particular those of the VirtualBox Host-Only Network adaptors #1 and #2. I modified the IPv4 properties . They had static IP addresses set (similar to those listed in the log above), so I changed them to "Obain an IP address Automatically". Then, when I ran ```otto compile```, it finished in milliseconds. The successful log didn't show anything about IP addresses or subnets.
